api/apps/geoprocessing/src/migrations/geoprocessing/1708524274000-AddStableIdsToFeaturesDataRows.ts

```diff
@@ -0,0 +1,44 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddStableIdsToFeaturesDataRows1708524274000
+  implements MigrationInterface
+{
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+DROP TRIGGER tr_precompute_feature_property_list ON features_data;
+
+CREATE TRIGGER tr_precompute_feature_property_list AFTER INSERT ON features_data
+FOR EACH ROW EXECUTE
+PROCEDURE precompute_feature_property_list();
+      `);
+
+    await queryRunner.query(`
+ALTER TABLE features_data
+ADD COLUMN stable_id uuid;
+
+CREATE INDEX features_data_stable_id__idx ON features_data(stable_id);
+
+CREATE UNIQUE INDEX features_data_unique_stable_ids_within_feature__idx ON features_data(feature_id, stable_id);
+      `);
+
+    await queryRunner.query(`
+UPDATE features_data
+SET stable_id = id;
+      `);
+
+    await queryRunner.query(`
+ALTER TABLE features_data
+ALTER COLUMN stable_id SET NOT NULL;
+
+ALTER TABLE features_data
+ALTER COLUMN stable_id SET DEFAULT gen_random_uuid();
```

Comment:
If I undesrtood correctly, it doesn't really matter, but to keep it consistent with the initial migration, this could be set to be the feature_data id, right? Unless it's actually not possible to reference a generated primary key in a another column's DEFAULT clause

Reply:
yep, as far as I know and could gather from PG's docs (https://www.postgresql.org/docs/current/sql-createtable.html#SQL-CREATETABLE-PARMS-DEFAULT), I think that this is not possible unless using a trigger, which I think would be overkill for this - I thought about this too because at least from fully new features (that is, not copied from a source project through cloning) it seems neater, even though in practice it doesn't matter.

Using a `generated always ... stored` column allows to reference other columns, but then it would not be possible to _set_ this column to the value we want. And using `generated ... identity` allows to update to user-supplied values, but this would apply to integer sequence columns only, so not good for our uuids. As far as I know, at least!